src/wp-includes/class-wp-theme.php

```diff
@@ -467,11 +494,12 @@ public function __construct( $theme_dir, $theme_root, $_child = null ) {
 					$this->cache_add(
 						'theme',
 						array(
-							'block_theme' => $this->is_block_theme(),
-							'headers'     => $this->headers,
-							'errors'      => $this->errors,
-							'stylesheet'  => $this->stylesheet,
-							'template'    => $this->template,
+							'block_theme'            => $this->is_block_theme(),
+							'block_template_folders' => $this->get_block_template_folders(),
```

Comment:
```suggestion
							'block_template_folders' => $this->get_block_template_folders(),
							'block_theme'            => $this->is_block_theme(),
```

Nit. Swap them like you did for otheres.